common/models/acl.js

```diff
@@ -455,18 +500,96 @@ module.exports = function(ACL) {
       async.parallel(inRoleTasks, function(err, results) {
         if (err) return callback(err, null);
 
+        // resolved is an instance of AccessRequest
         var resolved = self.resolvePermission(effectiveACLs, req);
-        if (resolved && resolved.permission === ACL.DEFAULT) {
-          resolved.permission = (model && model.settings.defaultPermission) || ACL.ALLOW;
-        }
+        resolved.settleDefaultPermission();
         debug('---Resolved---');
         resolved.debug();
+
+        // set authorizedRoles in remotingContext options argument if
+        // resolved AccessRequest permission is ALLOW, else set it to empty object
+        authorizedRoles = resolved.isAllowed() ? authorizedRoles : {};
+        saveAuthorizedRolesToRemotingContext(remotingContext, authorizedRoles);
         return callback(null, resolved);
       });
     });
     return callback.promise;
   };
 
+  function addRoleToAuthorizedRoles(role, authorizedRoles, context) {
+    let valueForAuthorizedRole = ACL.getValueForRoleInAuthorizedRoles;
+    if (typeof valueForAuthorizedRole !== 'function')
+      return;
+    authorizedRoles[role] = valueForAuthorizedRole(context);
+  }
+
```

Comment:
+ this should not be a class method as we do not want users to easily modify this code
+ it allows using a custom `valueForAuthorizedRole`method in case the user wants more advanced values for authorized roles